tensorboard/summary/output_test.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for the `tensorboard.summary._output` module."""
+
+
+import numpy as np
+
+from tensorboard.compat.proto import summary_pb2
+from tensorboard.summary import _output as output_lib
+from tensorboard.summary import test_util
+from tensorboard.util import tensor_util
+from tensorboard import test as tb_test
+
+
+class DirectoryOutputTest(tb_test.TestCase):
+    def test_empty(self):
+        logdir = self.get_temp_dir()
+        output = output_lib.DirectoryOutput(logdir)
+        output.close()
+        events = test_util.read_tfevents(logdir)
+        self.assertLen(events, 1)
+        self.assertEqual(events[0].file_version, "brain.Event:2")
+
+    def test_flush_and_close(self):
+        logdir = self.get_temp_dir()
+        output = output_lib.DirectoryOutput(logdir)
+        emit_scalar = lambda step: output.emit_scalar(
+            plugin_name="plugin",
+            tag="tag",
+            data=np.float32(0),
+            step=np.int64(step),
+            wall_time=0.0,
+            tag_metadata=b"meta",
+            description="desc",
+        )
+        emit_scalar(1)
+        events = test_util.read_tfevents(logdir)
+        # Only the file version event should be present.
+        self.assertLen(events, 1)
```

Comment:
Hmm, looks like this is failing in notf—not obvious to me why, but could
be genuine?

Reply:
Fixed by relaxing the test and leaving a comment, as discussed.